tensorboard/webapp/app_routing/internal_utils_test.ts

```diff
@@ -406,4 +406,67 @@ describe('app_routing/utils', () => {
       ).toBe(false);
     });
   });
+
+  describe('#getDeepLinkGroup', () => {
+    it('maps RouteKind to DeepLinkGroup', () => {
+      expect(utils.getDeepLinkGroup(RouteKind.EXPERIMENTS)).toEqual(
+        DeepLinkGroup.EXPERIMENTS
+      );
+      expect(utils.getDeepLinkGroup(RouteKind.EXPERIMENT)).toEqual(
+        DeepLinkGroup.DASHBOARD
+      );
+      expect(utils.getDeepLinkGroup(RouteKind.COMPARE_EXPERIMENT)).toEqual(
+        DeepLinkGroup.DASHBOARD
+      );
+      expect(utils.getDeepLinkGroup(RouteKind.UNKNOWN)).toBeNull();
+      expect(utils.getDeepLinkGroup(RouteKind.NOT_SET)).toBeNull();
+    });
+  });
+
+  describe('#canRehydrateDeepLink', () => {
+    it('allows rehydration if namespaceId does not match', () => {
+      expect(
+        utils.canRehydrateDeepLink(RouteKind.EXPERIMENTS, 'namespaceC', [
+          {
+            deepLinkGroup: DeepLinkGroup.EXPERIMENTS,
+            namespaceId: 'namespaceA',
+          },
+          {
+            deepLinkGroup: DeepLinkGroup.EXPERIMENTS,
+            namespaceId: 'namespaceB',
+          },
+        ])
+      ).toBeTrue();
+    });
+
+    it('allows rehydration if deepLinkGroup does not match', () => {
+      expect(
+        utils.canRehydrateDeepLink(RouteKind.COMPARE_EXPERIMENT, 'namespaceA', [
+          {
+            deepLinkGroup: DeepLinkGroup.EXPERIMENTS,
+            namespaceId: 'namespaceA',
+          },
+          {
+            deepLinkGroup: DeepLinkGroup.EXPERIMENTS,
+            namespaceId: 'namespaceB',
+          },
+        ])
+      ).toBeTrue();
+    });
+
+    it('does not allow rehydration if match is found', () => {
+      expect(
+        utils.canRehydrateDeepLink(RouteKind.EXPERIMENTS, 'namespaceA', [
+          {
+            deepLinkGroup: DeepLinkGroup.EXPERIMENTS,
+            namespaceId: 'namespaceA',
+          },
+          {
+            deepLinkGroup: DeepLinkGroup.EXPERIMENTS,
+            namespaceId: 'namespaceB',
+          },
+        ])
+      ).toBeFalse();
+    });
```

Comment:
add a test for routeKind is null?

Reply:
I added a test for when DeepLinkGroup is null (RouteKind cannot be null).